src/Entities/CourseEntity.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Portal\Entities;
+
+use phpDocumentor\Reflection\Types\This;
+
+class CourseEntity
+{
+    protected $startDate;
+    protected $endDate;
+    protected $name;
+    protected $trainer;
+    protected $notes;
+    protected $id;
+
+    public function __construct(
+        string $startDate = null,
+        string $endDate = null,
+        string $name = null,
+        string $trainer = null,
+        string $notes = null,
+        int $id = null
+    ) {
+        $this->startDate = ($this->startDate ?? $startDate);
+        $this->endDate = ($this->endDate ?? $endDate);
+        $this->name = ($this->name ?? $name);
+        $this->trainer = ($this->trainer ?? $trainer);
+        $this->notes = ($this->notes ?? $notes);
+        $this->id = ($this->id ?? $id);
```

Comment:
As I mentioned in a previous comment, I'd suggest adding a sanitse data method in here, like the applicant entity.

Reply:
This way the all of the data will be automatically sanitised when you try and put it into an entity